resources/azure/environmentDefaults.go

```diff
@@ -43,8 +46,11 @@ func sandboxDefault() environmentDefault {
 			defaultVNet:            "/subscriptions/8c56d827-5f07-45ce-8f2b-6c5001db5c6f/resourceGroups/datadog-agent-testing/providers/Microsoft.Network/virtualNetworks/default-vnet",
 			defaultSubnet:          "/subscriptions/8c56d827-5f07-45ce-8f2b-6c5001db5c6f/resourceGroups/datadog-agent-testing/providers/Microsoft.Network/virtualNetworks/default-vnet/subnets/default-subnet",
 			defaultSecurityGroup:   "/subscriptions/8c56d827-5f07-45ce-8f2b-6c5001db5c6f/resourceGroups/datadog-agent-testing/providers/Microsoft.Network/networkSecurityGroups/default",
-			defaultInstanceType:    "Standard_B4ms",
-			defaultARMInstanceType: "Standard_D4ps_v5",
+			defaultInstanceType:    "Standard_D4s_v3",  // Allows nested virtualization for kata runtimes
+			defaultARMInstanceType: "Standard_D4ps_v5", // No azure arm instance supports nested virtualization
+			defaultKataOsSku:       "Mariner",
+			defaultDeployKata:      true,
+			defaultKataRuntime:     "KataMshvVmIsolation", // Azure also supports KataCcIsolation
```

Comment:
`KataCcIsolation` requires a whole setup with OIDC and Identity mapping. None of that being implemented, I would not support this option.

Reply:
Hardcoded it in the last commit